test/blackbox_tests/dns_test.go

```diff
@@ -20,27 +20,29 @@ var _ = Describe("Outbound DNS/UDP traffic to port 53", func() {
 	var ns *netns.NetNS
 
 	BeforeEach(func() {
-		DeferCleanup(ns.Cleanup)
-
 		ns, err = netns.NewNetNSBuilder().Build()
 		Expect(err).To(BeNil())
 	})
 
+	AfterEach(func() {
```

Comment:
I was thinking we decided to go with Defer? :)

Reply:
We did, but it somehow was not reliable in these tests, and I didn't had enough time to debug why, as moving cleaning to AfterEach fixed all issues.

Reply:
ok, I know why - `DeferCleanup(ns.Cleanup)` is always being called on `nil` as it's "locking" the `ns` here